storage/embedded/src/main/java/one/microstream/storage/embedded/types/EmbeddedStorageObjectRegistryCallback.java

```diff
@@ -0,0 +1,76 @@
+package one.microstream.storage.embedded.types;
+
+import one.microstream.persistence.types.ObjectIdsProcessor;
+import one.microstream.persistence.types.ObjectIdsSelector;
+import one.microstream.persistence.types.PersistenceObjectRegistry;
+
+public interface EmbeddedStorageObjectRegistryCallback extends ObjectIdsSelector
+{
+	public void initializeObjectRegistry(PersistenceObjectRegistry objectRegistry);
+
+
+
+	public static EmbeddedStorageObjectRegistryCallback New()
+	{
+		return new EmbeddedStorageObjectRegistryCallback.Default();
+	}
+
+	public final class Default implements EmbeddedStorageObjectRegistryCallback
+	{
+		///////////////////////////////////////////////////////////////////////////
+		// instance fields //
+		////////////////////
+
+		private PersistenceObjectRegistry objectRegistry;
+
+
+
+		///////////////////////////////////////////////////////////////////////////
+		// constructors //
+		/////////////////
+
+		Default()
+		{
+			super();
+		}
+
+
+
+		///////////////////////////////////////////////////////////////////////////
+		// methods //
+		////////////
+
+		@Override
+		public synchronized void initializeObjectRegistry(final PersistenceObjectRegistry objectRegistry)
+		{
+			if(this.objectRegistry != null)
+			{
+				if(this.objectRegistry == objectRegistry)
+				{
+					return;
+				}
+
+				// (29.07.2022 TM)EXCP: proper exception
```

Comment:
is this a TODO?

Reply:
All places where RuntimeExceptions should be replaced by properly typed ones are marked with this expression. It is about time to replace them; see #403.